crates/uv-extract/src/error.rs

```diff
@@ -28,4 +28,12 @@ impl Error {
             Self::AsyncZip(async_zip::error::ZipError::FeatureNotSupported(_))
         )
     }
+
+    /// Returns `true` if the error is due to HTTP streaming request failed.
+    pub fn is_http_streaming_failed(&self) -> bool {
+        matches!(
+            self,
+            Self::AsyncZip(async_zip::error::ZipError::UpstreamReadError(_)) | Self::Io(_)
```

Comment:
The `Io` error case is for https://github.com/astral-sh/uv/blob/545bbf92863f4109070ebeb922c837b430441b76/crates/uv-extract/src/stream.rs#L53